services/brig/src/Brig/API/User.hs

```diff
@@ -1371,41 +1386,29 @@ lookupProfile self other =
 -- Otherwise only the 'PublicProfile' is accessible for user 'self'.
 -- If 'self' is an unknown 'UserId', return '[]'.
 lookupProfiles ::
-  ( MonadUnliftIO m,
-    MonadClient m,
-    MonadReader Env m,
-    MonadLogger m,
-    MonadMask m,
-    MonadHttp m,
-    HasRequestId m
-  ) =>
+  Members '[GalleyProvider] r =>
   -- | User 'self' on whose behalf the profiles are requested.
   Local UserId ->
   -- | The users ('others') for which to obtain the profiles.
   [Qualified UserId] ->
-  ExceptT FederationError m [UserProfile]
+  ExceptT FederationError (AppT r) [UserProfile]
+-- TODO(sandy): PERFORMANCE CHANGE: no longer concurrent
```

Comment:
Are you planning to address this as part of this PR? I believe it should be done. In my PR #2450 I need the same effect that interprets in `MonadUnliftIO`, but I am yet to write it.

Reply:
Sounds reasonable to me. What is unfortunate is there are no performance tests in Wire server, so we don't know how much this concurrency buys us or how much it would hurt the performance if it was removed.

@jschaul , do you have any insights into how impactful the change would be should this function become sequential instead of concurrent?

Reply:
Sounds reasonable to me. What is unfortunate is there are no performance tests in Wire server, so we don't know how much this concurrency buys us or how much it would hurt the performance if it was removed.

@jschaul , do you have any insights into how impactful the change would be should this function become sequential instead of concurrent?

Reply:
I've reimplemented this in terms of concurrency.